feat(ui): keyboard support and usability improvements for layer list panel

milesap opened the pull request:
Previously the layer list only expanded on mouseover. Now the layer list expands when tabbing through items. Focus is also placed on the panels close button when tabbing off the first or last item, or when an item is selected with the keyboard. When using a mouse, the layer list is expanded with a slight delay which is triggered when the mouse movement speed is slowed over an item. jquery-hoverIntent library implements this logic.

Closes #147

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/721)
<!-- Reviewable:end -->

alyec commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/ui/details/details.directive.spec.js, line 30 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfPbfMEDDy0GJzDTcD:-KJfPbfMEDDy0GJzDTcE:-744773829) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.spec.js#L30)):*
> ```JavaScript
>     });
> 
>     describe('rvDetails', () => {
> ```

Use `xdescribe` or `xit` to disable tests as these will show as disabled in the test run.  Try to include a comment about why the test is disabled instead of fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
*[src/app/ui/details/details-slider.directive.js, line 71 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfRQacWLziFQlnQvX0:-KJfRQacWLziFQlnQvX1:-226063297) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L71)):*
> ```JavaScript
> 
>             self.itemSelectedByKeypress = (evt, item, isLast) => {
>                 if (evt.which === 13 || evt.which === 32 || evt.which === 33) {
> ```

Don't need 33 (its `!`)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
Layer list wont open on mouse hover for me

<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> #### feat\(ui\): keyboard support and usability improvements for layer list panel
> 
> Previously the layer list only expanded on mouseover. Now the layer list expands when tabbing through items. Focus is also placed on the panels close button when tabbing off the first or last item, or when an item is selected with the keyboard. When using a mouse, the layer list is expanded with a slight delay which is triggered when the mouse movement speed is slowed over an item. jquery-hoverIntent library implements this logic.
> 
> 
> 
> Closes #147
> 
> 
> 
</details>


---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KJfbbBMQxfKGR6KV0XS:1038969197)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
Tab also seems to loop, layerlist -> close -> layerlist etc. etc.

<details><summary><i>Previously, spencerwahl (Spencer Wahl) wrote…</i></summary>
> Layer list wont open on mouse hover for me
</details>


---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KJfbmv4Fs-ChaWdfI2Y:2026347385)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
`Shift` + `Tab` on the last element in the layer selector goes to the close button, not the previous element.

<details><summary><i>Previously, spencerwahl (Spencer Wahl) wrote…</i></summary>
> Tab also seems to loop, layerlist -> close -> layerlist etc. etc.
</details>


---

Reviewed 4 of 7 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[src/app/common/router/statemanager.service.js, line 301 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfj0-ciTr1F_vyLwKv:-KJfj0-ciTr1F_vyLwKw:1615979290) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/common/router/statemanager.service.js#L301)):*
> ```JavaScript
>          * Changes focus to the last saved focusable element
>          */
>         function nextFocus() {
> ```

Misleading name - `nextFocus` sets focus to `previous` element.

---

*[src/app/ui/details/details-slider.directive.js, line 23 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfjkcRdFnXwAIXFWLP:-KJfjkcRdFnXwAIXFWLQ:271441207) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L23)):*
> ```JavaScript
>         .directive('rvDetailSlider', rvDetailSlider);
> 
>     function rvDetailSlider(stateManager) {
> ```

Please, jsdoc-comment you functions and add comments where needed to explain what is going on.

---

*[src/app/ui/details/details-slider.directive.js, line 42 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfl8OogvHU-jJv6-4L:-KJfl8Op5Ccuz-36Utlm:-179595557) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L42)):*
> ```JavaScript
>             let forceClose = false;
> 
>             const animateOpen = () => {
> ```

Why is this a variable?

---

*[src/app/ui/details/details-slider.directive.js, line 52 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffmrH3xwdvKqBrg62:-KJffmrIHibVSdVUz0FJ:1873182874) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L52)):*
> ```JavaScript
>             };
> 
>             const animateClosed = () => tl.reversed(true);
> ```

Why is this a variable?
Why not just use `reverse`?

---

*[src/app/ui/details/details.directive.js, line 22 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfdXwXEocRtYHHu2he:-KJfdXwXEocRtYHHu2hf:1530771804) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.js#L22)):*
> ```JavaScript
>             restrict: 'E',
>             templateUrl: 'app/ui/details/details.html',
>             link
> ```

Link function are for DOM manipulations mostly. Everything else should go into the controller.

---

*[src/app/ui/details/details.directive.js, line 68 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffMX-Q6jG8ooDA_5_:-KJffMX-Q6jG8ooDA_5a:1449316100) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L68)):*
> ```JavaScript
>                         // This will explicitly "animate" the overflow property from hidden to auto and not try to figure
>                         // out what it was initially on the reverse run.
>                         .fromTo(layerListNode, 0.01, {
> ```

Any reason to remove this block? It helps to hide the scrollbar during the closing animation.

---

*[src/app/ui/details/details.directive.js, line 115 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfd03fyZFXxzPoT0dR:-KJfd03fyZFXxzPoT0dS:-1289494230) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L115)):*
> ```JavaScript
>         /*********/
> 
>         /**
> ```

Any reason why you are removing this?

---

*[src/app/ui/details/details.html, line 37 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfckXvb-IkZX7pNYTE:-KJfckXvb-IkZX7pNYTF:-1925124025) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.html#L37)):*
> ```HTML
>         <!-- the detail content  -->
>         <!-- the side selector  -->
>         <rv-detail-slider class="rv-details-layer-list" ng-show="self.display.data.length > 1"></rv-details-slider>
> ```

Is this `ng-show` really needed? Also, `detail-slider` is not very descriptive name.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KJfjYrE0oUdi5RPfZEQ:2050988080)*
<!-- Sent from Reviewable.io -->

milesap commented:
:+1: 

<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> `Shift` + `Tab` on the last element in the layer selector goes to the close button, not the previous element.
</details>


---

Review status: 0 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[src/app/common/router/statemanager.service.js, line 301 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfj0-ciTr1F_vyLwKv:-KJj_AwVvK30mceZM2c3:1826121865) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/common/router/statemanager.service.js#L301)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Misleading name - `nextFocus` sets focus to `previous` element.
</details>
:+1: changed to previousFocus

---

*[src/app/ui/details/details.directive.js, line 22 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfdXwXEocRtYHHu2he:-KJjbzb3BSmPFNJTPbpx:-1224885100) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.js#L22)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Link function are for DOM manipulations mostly. Everything else should go into the controller.
</details>
I noticed this in the angular documentation:
Best Practice: use controller when you want to expose an API to other directives. Otherwise use link.
https://docs.angularjs.org/guide/directive

Since this directive has no need for an exposed API a link seemed like a better option.

---

*[src/app/ui/details/details.directive.js, line 68 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffMX-Q6jG8ooDA_5_:-KJjf0DBhxiDmLOBityx:1447449173) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L68)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Any reason to remove this block? It helps to hide the scrollbar during the closing animation.
</details>
Added back - I didn't notice a difference with having this set, but it doesn't hurt to keep it

---

*[src/app/ui/details/details.directive.js, line 115 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfd03fyZFXxzPoT0dR:-KJjg-YNPDyqBGLoCwR-:-1184774401) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L115)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Any reason why you are removing this?
</details>
:+1:  restored getSelectedItem function

---

*[src/app/ui/details/details.directive.spec.js, line 30 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfPbfMEDDy0GJzDTcD:-KJjks_1P-EBP8ZljjIj:-625482228) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.spec.js#L30)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> Use `xdescribe` or `xit` to disable tests as these will show as disabled in the test run.  Try to include a comment about why the test is disabled instead of fixed.
</details>
:+1:  good to know

---

*[src/app/ui/details/details.html, line 37 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfckXvb-IkZX7pNYTE:-KJjjWlTCPtakENsx_Zx:1864825555) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.html#L37)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Is this `ng-show` really needed? Also, `detail-slider` is not very descriptive name.
</details>
:+1: removed ng-show and made more descriptive

---

*[src/app/ui/details/details-slider.directive.js, line 23 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfjkcRdFnXwAIXFWLP:-KJja4HY4OZ-ghi9ZjW5:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L23)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Please, jsdoc-comment you functions and add comments where needed to explain what is going on.
</details>
:+1: 

---

*[src/app/ui/details/details-slider.directive.js, line 42 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfl8OogvHU-jJv6-4L:-KJjaCaMi1VnzOQ6bE-Z:1366489932) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L42)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Why is this a variable?
</details>
changed to function

---

*[src/app/ui/details/details-slider.directive.js, line 52 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffmrH3xwdvKqBrg62:-KJjaHaipJFDDuwzn8Di:-714144422) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L52)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Why is this a variable?
> Why not just use `reverse`?
</details>
changed to a function. I prefer reversed(boolean) personally as it explicitly sets the animation direction. reverse can introduce errors where reverse on a reversed animation moves it forward.

---

*[src/app/ui/details/details-slider.directive.js, line 71 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfRQacWLziFQlnQvX0:-KJjaxqRik54HVliE9GJ:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details-slider.directive.js#L71)):*
<details><summary><i>Previously, spencerwahl (Spencer Wahl) wrote…</i></summary>
> Don't need 33 (its `!`)
</details>
:+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KJj_DEUq2PwH3iDq2cv:-510466975)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 7 of 10 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/app/ui/details/details.directive.js, line 22 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfdXwXEocRtYHHu2he:-KJkE7qTrtlTfu4QaNwf:-1395830855) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.js#L22)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> I noticed this in the angular documentation:
> Best Practice: use controller when you want to expose an API to other directives. Otherwise use link.
> https://docs.angularjs.org/guide/directive
> 
> Since this directive has no need for an exposed API a link seemed like a better option.
</details>
We follow different guidelines. For consistency, please, stick to DOM manipulations only in a link function, everything else should be put in a controller (it's already set up this way). 

---

*[src/app/ui/details/details.directive.js, line 68 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffMX-Q6jG8ooDA_5_:-KJkEnXlsbRsXwe5NeZH:-1188276951) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L68)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Added back - I didn't notice a difference with having this set, but it doesn't hurt to keep it
</details>
It doesn't mean the difference is not there. Please, don't just remove code you don't understand. If in doubt, ask.

---

*[src/app/ui/details/details.directive.js, line 38 \[r2\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJkFj_9PgghDIVgNirB:-KJkFj_AhfblBwSV9VTx:-1537075885) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/db53be1414248a1dab2965e96919b233e9678109/src/app/ui/details/details.directive.js#L38)):*
> ```JavaScript
>              * @param  {Object} item data object
>              */
>             self.selectItem = item => {
> ```

Please, for consistency, stick to existing way of adding functions: declaring functions at the bottom and assigning them at the top of a controller/link function.

---

*[src/app/ui/details/layer-list-slider.directive.js, line 52 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffmrH3xwdvKqBrg62:-KJkHJlMbaW2KJgUtdDm:372181135) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/layer-list-slider.directive.js#L52)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> changed to a function. I prefer reversed(boolean) personally as it explicitly sets the animation direction. reverse can introduce errors where reverse on a reversed animation moves it forward.
</details>
Calling reverse on an animation playing in reverse does not cause it to start playing forwards again.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 5 of 8 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/app/ui/details/details.directive.js, line 22 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJfdXwXEocRtYHHu2he:-KJtwhdUEo4hMsqCGgIk:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/details.directive.js#L22)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> We follow different guidelines. For consistency, please, stick to DOM manipulations only in a link function, everything else should be put in a controller (it's already set up this way). 
</details>
:+1: 

---

*[src/app/ui/details/details.directive.js, line 68 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffMX-Q6jG8ooDA_5_:-KJtwjMPoPkVcVJLOLRS:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/40e41e25da7c9e0f62c5b0eeb8d8e6e3f541bdac/src/app/ui/details/details.directive.js#L68)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> It doesn't mean the difference is not there. Please, don't just remove code you don't understand. If in doubt, ask.
</details>
:+1: 

---

*[src/app/ui/details/details.directive.js, line 38 \[r2\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJkFj_9PgghDIVgNirB:-KJtxOj1sbin_G6U6RZn:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/db53be1414248a1dab2965e96919b233e9678109/src/app/ui/details/details.directive.js#L38)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Please, for consistency, stick to existing way of adding functions: declaring functions at the bottom and assigning them at the top of a controller/link function.
</details>
:+1: 

---

*[src/app/ui/details/layer-list-slider.directive.js, line 52 \[r1\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJffmrH3xwdvKqBrg62:-KJtxR0cI5bIMpxSAkJH:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/61152fb61c569264e6b87ef910f8e828a1d50a43/src/app/ui/details/layer-list-slider.directive.js#L52)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Calling reverse on an animation playing in reverse does not cause it to start playing forwards again.
</details>
:+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

alyec commented:
`chore(ui)` -> `refactor(ui)` in the commit comments

<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> :+1: 
</details>


---

Reviewed 1 of 10 files at r2, 1 of 1 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KJuFKfm3HIE6fv9CZJJ:684281058)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
Reviewed 7 of 10 files at r2, 1 of 1 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[src/app/ui/details/details.directive.js, line 73 \[r6\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJvUcNa4-hjjuZSbSrK:-KJvUcNa4-hjjuZSbSrL:284500599) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/b428b2d8ec4bdf120f82fadddbb4313849ef2b2a/src/app/ui/details/details.directive.js#L73)):*
> ```JavaScript
>                             'overflow-y': 'auto'
>                         }, RV_SLIDE_DURATION / 2)
>                         .to(sectionNode, RV_SLIDE_DURATION, {
> ```

we need to carry this over as well, only problem is we need `sectionNode` available in the child directive. I think you should be able to pass it down on the scope and access it from the slider directive.

---

*[src/app/ui/details/layer-list-slider.directive.js, line 45 \[r6\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJv9DhYzlkFe9KQHVMj:-KJv9DhYzlkFe9KQHVMk:1666209934) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1462d86ac23dfee7f73eac726a11580a10dc8c83/src/app/ui/details/layer-list-slider.directive.js#L45)):*
> ```JavaScript
>                 width: 280,
>                 ease: RV_SWIFT_IN_OUT_EASE,
>                 overflowY: 'auto'
> ```

we dont need this line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 10 files at r2, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 2 of 10 files at r2, 1 of 1 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/ui/details/details.directive.js, line 73 \[r6\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJvUcNa4-hjjuZSbSrK:-KK9J0r_B7Gk8xiEGTsq:-157231737) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/b428b2d8ec4bdf120f82fadddbb4313849ef2b2a/src/app/ui/details/details.directive.js#L73)):*
<details><summary><i>Previously, spencerwahl (Spencer Wahl) wrote…</i></summary>
> we need to carry this over as well, only problem is we need `sectionNode` available in the child directive. I think you should be able to pass it down on the scope and access it from the slider directive.
</details>
You still need this to set the `rv-expanded` classname to the `sectionNode`:
```js
.to(sectionNode, RV_SLIDE_DURATION, {
    className: '+=rv-expanded'
}, 0);
```

---

*[src/app/ui/details/layer-list-slider.directive.js, line 45 \[r6\]](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-KJv9DhYzlkFe9KQHVMj:-KK9IQMTCi7OMkjVVzHY:-510466975) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1462d86ac23dfee7f73eac726a11580a10dc8c83/src/app/ui/details/layer-list-slider.directive.js#L45)):*
<details><summary><i>Previously, spencerwahl (Spencer Wahl) wrote…</i></summary>
> we dont need this line
</details>
:+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Just try to avoid watches in the future.

<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> `chore(ui)` -> `refactor(ui)` in the commit comments
</details>


---

Review status: 6 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721#-:-KKE3BM00X517hOvDmgP:-1616859478)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 1 of 10 files at r2, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->

spencerwahl commented:
Reviewed 1 of 10 files at r2, 2 of 2 files at r8.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/721)*
<!-- Sent from Reviewable.io -->
